pkg/controller/elasticsearch/driver/upgrade_deletion_strategy.go

```diff
@@ -0,0 +1,266 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Same here, file name does not reflect the current structure of the code  as you removed the strategy abstraction.